src/protocols/user_directed_commissioning/UserDirectedCommissioningClient.cpp

```diff
@@ -41,19 +43,32 @@ CHIP_ERROR UserDirectedCommissioningClient::SendUDCMessage(TransportMgrBase * tr
     }
     ChipLogProgress(Inet, "Sending UDC msg");
 
-    // send UDC message 5 times per spec (no ACK on this message)
-    for (unsigned int i = 0; i < 5; i++)
+    mUdcAttemptsRemaining = 5; // send UDC message 5 times per spec (no ACK on this message)
+    mTransportMgr         = transportMgr;
+    mTargetPayload        = payload.CloneData();
+    mTargetPeerAddress    = peerAddress;
+    DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds32(0), SendUDCTask, this);
+
+    return err;
+}
+
+void UserDirectedCommissioningClient::SendUDCTask(System::Layer * aSystemLayer, void * aAppState)
+{
+    UserDirectedCommissioningClient * thiz = reinterpret_cast<UserDirectedCommissioningClient *>(aAppState);
+
+    // Schedule next UDC msg, if any, 100ms later
+    thiz->mUdcAttemptsRemaining--;
+    if (thiz->mUdcAttemptsRemaining > 0)
     {
-        err = transportMgr->SendMessage(peerAddress, payload.CloneData());
-        if (err != CHIP_NO_ERROR)
-        {
-            ChipLogError(AppServer, "UDC SendMessage failed: %" CHIP_ERROR_FORMAT, err.Format());
-            return err;
-        }
-        usleep(100 * 1000); // 100ms
+        DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds32(100), SendUDCTask, thiz);
```

Comment:
nit, might be nice for 100 to be a variable somewhere

Reply:
Done